types/react-modal-video/index.d.ts

```diff
@@ -0,0 +1,118 @@
+// Type definitions for react-modal-video 1.2
+// Project: https://github.com/appleple/react-modal-video
+// Definitions by: Julián Bovone <https://github.com/jbovone>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.1
+
+import * as React from 'react';
+import { VimeoApi, YoukuApi, YoutubeApi } from './vendors';
+
+interface Vendor {
+    /**
+     * @default "youtube"
+     */
+    channel: string;
+    videoId: string;
+}
+interface Youku extends Vendor {
+    channel: 'youku';
+    youku?: YoukuApi;
+}
+interface Youtube extends Vendor {
+    channel: 'youtube';
+    /**
+     * @link https://developers.google.com/youtube/player_parameters
+     */
+    youtube?: YoutubeApi;
+}
+interface Vimeo extends Vendor {
+    channel: 'vimeo';
+    /**
+     * @link https://developer.vimeo.com/api/oembed/videos
+     */
+    vimeo?: VimeoApi;
+}
+interface Custom {
+    channel: 'custom';
+    url: string;
+}
+
+export interface ClassNames {
+    modalVideoEffect: string;
+    modalVideo: string;
+    modalVideoClose: string;
+    modalVideoBody: string;
+    modalVideoInner: string;
+    modalVideoIframeWrap: string;
+    modalVideoCloseBtn: string;
+}
+export interface Aria {
+    openMessage: string;
+    dismissBtnMessage: string;
+}
+export interface ModalVideoBaseProps {
+    /**
+     * @default false
+     */
+    isOpen: boolean;
+    onClose?: () => void;
+    /**
+     * @default true
+     */
+    allowFullScreen: boolean;
+    /**
+     * @default '16:9'
+     */
+    ratio: `${number}:${number}`;
+    /**
+     * @default 300
+     */
+    animationSpeed: number;
+    /**
+     * @default
+     *  {
+     *       modalVideo: 'modal-video',
+     *       modalVideoClose: 'modal-video-close',
+     *       modalVideoBody: 'modal-video-body',
+     *       modalVideoInner: 'modal-video-inner',
+     *       modalVideoIframeWrap: 'modal-video-movie-wrap',
+     *       modalVideoCloseBtn: 'modal-video-close-btn'
+     *   }
+     */
+    classNames: ClassNames;
+    /**
+     * @default
+     * {
+     *      openMessage: 'You just opened the modal video';
+     *       dismissBtnMessage: 'Close the modal by clicking here';
```

Comment:
```suggestion
     *      dismissBtnMessage: 'Close the modal by clicking here';
```

Reply:
So you actually prefer reflect always reflect the source over own modularization, ok that's fine.

Honestly, I'm not sure about keeping those... I mean there is certain value in having them, but Im not positive that they can be properly maintained.

For instance:
> This parameter specifies whether the initial video will automatically start to play when the player loads. Supported values are 0 or 1. The default value is 0.

I just figure it out (in a git-hub issue yesterday) that you also need to set mute=1, that's not specified in the docs.
And you have this wmode field in the source code, but is not in the docs.

They can cause problems or offer misleading information because changes overtime. The Api is volatile, not versioned, and the docs aren't perfect.